src/main/java/training/busboard/UserInterface.java

```diff
@@ -0,0 +1,88 @@
+package training.busboard;
+
+import training.busboard.models.Bus;
+
+import java.util.InputMismatchException;
+import java.util.List;
+import java.util.Scanner;
+
+public class UserInterface {
+    private Scanner scanner = new Scanner(System.in);
+
+    public void welcome() {
+        System.out.println("Welcome to BusBoard!!");
+        System.out.println("=====================");
+    }
+
+    public String getServiceToRun() {
+        System.out.println("Would you like to search for buses by postcode (enter p) or by stopcode (enter s)?");
+        System.out.print("Enter /q/ to quit. > ");
+        String input = scanner.nextLine();
+        try {
+            return this.processInput(input, "[sp]");
+        } catch (InputMismatchException e) {
+            System.out.println("You have entered an invalid postcode. Please try again.");
+            return this.getPostcode();
+        }
+    }
+
+    public String getStopcode() {
+        System.out.print("Please enter a stop code or enter /q/ to quit. > ");
+        String input = scanner.nextLine().trim().toLowerCase();
+        try {
+            return this.processInput(input, "\\d+\\w*");
+        } catch (InputMismatchException e) {
+            System.out.println("You have entered an invalid stopcode. Please try again.");
+            return this.getStopcode();
+        }
+    }
+
+    public String getPostcode() {
+        System.out.print("Please enter a postcode or enter /q/ to quit. > ");
+        String input = scanner.nextLine().trim().toLowerCase();
+
+        try {
+            return this.processInput(input, "\\w{1,2}\\d{1,2}\\s+\\d\\w\\w");
+        } catch (InputMismatchException e) {
+            System.out.println("You have entered an invalid postcode. Please try again.");
+            return this.getPostcode();
+        }
+    }
+
+    private String processInput(String input, String regex) throws InputMismatchException {
```

Comment:
https://docs.oracle.com/javase/7/docs/api/java/util/Scanner.html#next(java.util.regex.Pattern) might do a lot of what this method does, but given you're also doing quit handling here maybe it's not quite a drop-in replacement.